Madness/String.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  String.swift
+//  Madness
+//
+//  Created by Josh Vera on 10/19/15.
+//  Copyright © 2015 Rob Rix. All rights reserved.
+//
+
+import Foundation
+
+public typealias CharacterParser = Parser<String.CharacterView, Character>.Function
+public typealias StringParser = Parser<String.CharacterView, String>.Function
+public typealias DoubleParser = Parser<String.CharacterView, Double>.Function
+
+public func digit() -> CharacterParser {
+	return oneOf("123456789")
+}
+
+public func double() -> DoubleParser {
+	let minus = (%"-")|?
+	let digits = digit()+
+	
+	return minus >>- { sign in
+		digits >>- { digits in
+			let double = digits.reduce(0.0, combine: { num, char in
+				return 10 * num + Double(String(char))!
+			})
+
+			return pure(sign != nil ? double : -1 * double)
+		}
+	}
+}
```

Comment:
- Should this use `strtod` to convert to `Double`?
- If not, should this handle scientific notation like `strtod` does?